feature/experimental/gribi/ate_tests/supervisor_failure/Supervisor_Failure_test.go

```diff
@@ -0,0 +1,347 @@
+			standbyRP = supervisor
+		} else if role == activeController {
+			activeRP = supervisor
+		} else {
+			t.Fatalf("Expected controller %s to be active or standby, got %v", supervisor, role)
+		}
+	}
+	if standbyRP == "" || activeRP == "" {
+		t.Fatalf("Expected non-empty activeRP and standbyRP, got activeRP: %v, standbyRP: %v", activeRP, standbyRP)
+	}
+	t.Logf("Detected activeRP: %v, standbyRP: %v", activeRP, standbyRP)
+
+	return standbyRP, activeRP
+}
+
+func TestSupFailure(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	ctx := context.Background()
+
+	// Configure the DUT
+	configureDUT(t, dut)
+
+	// Configure the ATE
+	ate := ondatra.ATE(t, "ate")
+	top := configureATE(t, ate)
+	top.Push(t).StartProtocols(t)
+
+	// Configure the gRIBI client clientA
+	clientA := gribi.Client{
+		DUT:                  dut,
+		FibACK:               false,
+		Persistence:          true,
+		InitialElectionIDLow: 10,
+	}
+	defer clientA.Close(t)
+	if err := clientA.Start(t); err != nil {
+		t.Fatalf("gRIBI Connection can not be established")
+	}
+
+	args := &testArgs{
+		ctx:     ctx,
+		clientA: &clientA,
+		dut:     dut,
+		ate:     ate,
+		top:     top,
+	}
+	// Program a route and ensure AFT telemetry returns FIB_PROGRAMMED
+	routeInstall(ctx, t, args)
+	supervisors := findComponentsByType(t, dut, controlcardType)
+	t.Logf("Found supervisor list: %v", supervisors)
+	// Only perform the switchover for the chassis with dual RPs/Supervisors.
+	if len(supervisors) != 2 {
+		t.Skipf("Dual RP/SUP is required on %v: got %v, want 2", dut.Model(), len(supervisors))
+	}
+
+	rpStandbyBeforeSwitch, rpActiveBeforeSwitch := findStandbyRP(t, dut, supervisors)
+	t.Logf("Detected rpStandby: %v, rpActive: %v", rpStandbyBeforeSwitch, rpActiveBeforeSwitch)
+
+	gnoiClient := dut.RawAPIs().GNOI().Default(t)
+	switchoverRequest := &spb.SwitchControlProcessorRequest{
+		ControlProcessor: &tpb.Path{
+			Elem: []*tpb.PathElem{{Name: rpStandbyBeforeSwitch}},
+		},
+	}
+	t.Logf("switchoverRequest: %v", switchoverRequest)
+	switchoverResponse, err := gnoiClient.System().SwitchControlProcessor(context.Background(), switchoverRequest)
+	if err != nil {
+		t.Fatalf("Failed to perform control processor switchover with unexpected err: %v", err)
+	}
+	t.Logf("gnoiClient.System().SwitchControlProcessor() response: %v, err: %v", switchoverResponse, err)
+
+	rpStandbyAfterSwitch, rpActiveAfterSwitch := findStandbyRP(t, dut, supervisors)
+	t.Logf("Found standbyRP after switchover: %v, activeRP: %v", rpStandbyAfterSwitch, rpActiveAfterSwitch)
+	t.Log("Validate OC Switchover time/reason.")
+	activeRP := dut.Telemetry().Component(rpActiveAfterSwitch)
+	if !activeRP.LastSwitchoverTime().Lookup(t).IsPresent() {
+		t.Errorf("activeRP.LastSwitchoverTime().Lookup(t).IsPresent(): got false, want true")
+	} else {
+		t.Logf("Found activeRP.LastSwitchoverTime(): %v", activeRP.LastSwitchoverTime().Get(t))
+	}
+
+	if !activeRP.LastSwitchoverReason().Lookup(t).IsPresent() {
```

Comment:
Similarly we would want to validate the value of switchover-reason, and not just whether it is present.

Reply:
I think it is already added.

                lastSwitchoverReason := activeRP.LastSwitchoverReason().Get(t)
                t.Logf("Found lastSwitchoverReason.GetDetails(): %v", lastSwitchoverReason.GetDetails())
                t.Logf("Found lastSwitchoverReason.GetTrigger().String(): %v", lastSwitchoverReason.GetTrigger().String())

Reply:
`lastSwitchoverReason.GetTrigger()` could be `telemetry.PlatformTypes_ComponentRedundantRoleSwitchoverReasonTrigger_SYSTEM_INITIATED` or
`telemetry.PlatformTypes_ComponentRedundantRoleSwitchoverReasonTrigger_USER_INITIATED`. 

In this specific case, it probably should be `USER_INITIATED`.

Reply:
Implemented.. New code push failed due to "Needs /gcbrun from a collaborator"..